guides/Selenium Tests Documentation.md

```diff
@@ -0,0 +1,326 @@
+
+-   `String browser` – An identifier for the browser the test is to be executed on (one of "firefox", "chrome", "ie" or "edge")
+-   `String rootUrl` – The root of the documentation hierarchy
+-   `String page` –  The relative location from the root of the documentation hierarchy of the web page to be tested. The complete URL will be constructed from the rootUrl and page parameters
+-   `String startingUrl` – May be used to enforce the page to be tested. If not specified the page's URL will be constructed from the `rootUrl` and `page` parameters
+-   `Integer implicitWaitTimeMS` – The maximum time, in milliseconds, to be taken to find an element or elements (default: 5000ms)
+-   `Integer displayWaitTimeMS` – The maximum time, in milliseconds, to wait to allow a page to be loaded before performing the next action
+    (default: 5000ms)
+-   `Integer verticalScrollPixels` - The number of pixels by which the window may be scrolled vertically to make controls visible (default -50)
+-   `String chromeDriverLocation` – The location on disk of the ChromeDriver executable, used to send commands to a Chrome browser
+-   `String geckoDriverLocation` – The location on disk of the GeckoDriver executable, used to send commands to a Firefox browser
+-   `String IEDriverLocation` – The location on disk of the IEDriveDriver executable, used to send commands to an IE 11 browser
+-   `String edgeDriverLocation` – The location on disk of the MicrosoftWebDriver executable, used to send commands to an Edge browser
+-   `Boolean useRemoteWebDriver` - Whether to send test commands to a Selenium Grid hub (true) or a local web driver (false) (default: true)
+-   `String browserPlatform` - The OS on which the browser being tested is running (one of "vista" (for Windows Vista, Windows Server 2008 and Windows 7), "win8" (for WIndows 8 and Windows Server 2012) or "win10" (for Windows 10))
+-   `String gridHubURL` - The URL of the Selenium Grid hub
+
+The class also contains methods which:
+
+-   record the values of input parameters for a test and create an instance of the relevant browser driver
+-   empty the list of errors before each method is executed
+-   perform tests on elements and log any reported errors
+-   set the test’s status to failed if errors occurred
+-   close the instance of the driver when all of a class’s methods have executed
+
+###### Utilities class
+
+The Utilities class contains methods which:
+
+-   determine the name of the currently executing method
+-   determine the line number of a method at which an error was raised
+-   use the above methods to log information about the location of an
+    error
+-   scroll to a documentation section
+-   scroll the browser screen vertically by a specified number of pixels
+    (to ensure a component is visible in the browser window)
+
+# Problems encountered during test development
+
+#### Component visibility in Chrome
+
+When a component such as a button is clicked through the WebDriver API,
```

Comment:
Ran tests on Chrome without using this adjustment. All tests worked.
This section has been removed from the document.
Code will be updated in a separate JIRA.